### PR TITLE
[RLlib] Bump gymnasium to 0.27.1

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

This PR bumps gymnasium to the latest release. This is needed because, by default, atari environments have uint8 spaces that can "slip through cracks" and end up in torch tensors with char types that are incompatible with NNs that have float variables.

This has surfaced in https://github.com/ray-project/ray/pull/32650 and is also needed for that PR.

### CODE REVIEW

File: rllib/models/catalog.py [hunk 1]
```diff
@@ -345,7 +345,7 @@ def get_action_dist(
                 "Unsupported args: {} {}".format(action_space, dist_type)
             )
 
-        return dist_cls, dist_cls.required_model_output_shape(action_space, config)
+        return dist_cls, int(dist_cls.required_model_output_shape(action_space, config))
```
Comment (ArturNiederfahrenhorst) [inline rllib/models/catalog.py:348]: This seems unrelated but is due to the fact that we calculate the output shape from spaces that may contain non Python int values, but rather np.int64 values ->

This commit in gymnasium: https://github.com/Farama-Foundation/Gymnasium/pull/141/files#diff-08f7c040be788bbfb4790fc76ac8837b74532e4977e4ef5d444d7b008a029e00R45
causes this issue for us.
So I'm proposing this change to force that we return the type promised in this method's signature.

File: rllib/env/wrappers/atari_wrappers.py [hunk 2]
```diff
@@ -27,7 +27,14 @@ def is_atari(env: Union[gym.Env, str]) -> bool:
...
             return False
-        return "AtariEnv<ALE" in str(env)
+        if (
+            hasattr(env, "spec")
+            and hasattr(env.spec, "kwargs")
+            and env.spec.kwargs.get("env_id").startswith("ALE/")
+        ):
+            return True
+        else:
+            return False
```
Comment (ArturNiederfahrenhorst) [inline rllib/env/wrappers/atari_wrappers.py:37]: `str(env)` has changed and does not include "ALE/" anymore after this change.
This change utilizes a similar "trick" to what we do in L40.

File: python/requirements.txt [hunk 3]
```diff
@@ -43,7 +43,7 @@ scikit-image
 scipy
 aiohttp>=3.7
 fastapi
-gymnasium==0.26.3
+gymnasium==0.27.1
```
Comment (araffin) [inline python/requirements.txt:46]: latest release is now 0.28.1

Comment (ArturNiederfahrenhorst) [reply python/requirements.txt:46]: Thanks!

**Comment by ArturNiederfahrenhorst** [comment]
Closing this in favour of https://github.com/ray-project/ray/pull/35698
Thanks @Rohan138

**Pull request closed without merging**